controller/contour/controller.go

```diff
@@ -128,66 +130,83 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 // ensureContour ensures all necessary resources exist for the given contour.
 func (r *Reconciler) ensureContour(ctx context.Context, contour *operatorv1alpha1.Contour) error {
+	var errs []error
 	if err := r.ensureNamespace(ctx, contour); err != nil {
-		return fmt.Errorf("failed to ensure namespace %s for contour %s/%s: %w",
-			contour.Spec.Namespace.Name, contour.Namespace, contour.Name, err)
+		errs = append(errs, fmt.Errorf("failed to ensure namespace %s for contour %s/%s: %w",
+			contour.Spec.Namespace.Name, contour.Namespace, contour.Name, err))
 	}
 	if err := r.ensureRBAC(ctx, contour); err != nil {
-		return fmt.Errorf("failed to ensure rbac for contour %s/%s: %w", contour.Namespace, contour.Name, err)
+		errs = append(errs, fmt.Errorf("failed to ensure rbac for contour %s/%s: %w", contour.Namespace, contour.Name, err))
 	}
-	if err := r.ensureConfigMap(ctx, contour); err != nil {
-		return fmt.Errorf("failed to ensure configmap for contour %s/%s: %w", contour.Namespace, contour.Name, err)
-	}
-	if err := r.ensureJob(ctx, contour); err != nil {
-		return fmt.Errorf("failed to ensure job for contour %s/%s: %w", contour.Namespace, contour.Name, err)
-	}
-	if err := r.ensureDeployment(ctx, contour); err != nil {
-		return fmt.Errorf("failed to ensure deployment for contour %s/%s: %w", contour.Namespace, contour.Name, err)
-	}
-	if err := r.ensureDaemonSet(ctx, contour); err != nil {
-		return fmt.Errorf("failed to ensure daemonset for contour %s/%s: %w", contour.Namespace, contour.Name, err)
-	}
-	if err := r.ensureContourService(ctx, contour); err != nil {
-		return fmt.Errorf("failed to ensure service for contour %s/%s: %w", contour.Namespace, contour.Name, err)
+
+	deploy := &appsv1.Deployment{}
+	ds := &appsv1.DaemonSet{}
```

Comment:
Tiny suggestion:
```suggestion
	var deploy appsv1.Deployment
	var ds appsv1.DaemonSet
```
now that `syncContourStatus` checks for nil values.

Reply:
Changing the vars as you suggest are an issue with `ensureDeployment(ctx, contour)` and `ensureDaemonSet(ctx, contour)`. Both funcs return pointers for these vars instead of values.

Reply:
@Miciah should I update these func to return values instead of pointers?

Reply:
> Changing the vars as you suggest are an issue with `ensureDeployment(ctx, contour)` and `ensureDaemonSet(ctx, contour)`. Both funcs return pointers for these vars instead of values.

Ah, sorry, I meant this:
```go

> @Miciah should I update these func to return values instead of pointers?